File: server/scripts/uploadData.js
const { db, auth } = require("../firebase/firebase");
const { addRecipes } = require("./generateRecipes");
const { addUser } = require("./generateUsers");

const uploadDataToDatabase = async () => {
  try {
    const recipesRef = db.collection("recipes");
    const users = addUser(5);

    if (!Array.isArray(users) || users.length === 0) {
      console.log("No users generated");
    }

    for (const user of users) {
      const userAuth = await auth.createUser({
        email: user.email,
        password: user.password,
        displayName: user.username,
      });

      await db.collection("users").doc(userAuth.uid).set({
        username: user.username,
        email: user.email,
        createdAt: user.createdAt,
      });

      const recipes = addRecipes(5, user.username);

      if (!Array.isArray(recipes) || recipes.length === 0) {
        console.log("No recipes generated");
        continue;
      }

      const batch = db.batch();

      for (const recipe of recipes) {
        const recipeRef = recipesRef.doc();
        batch.set(recipeRef, recipe);
      }

      await batch.commit();
    }
  } catch (error) {
    console.log(error.message);
  }
};

module.exports = {
  uploadDataToDatabase,
};

File: server2/scripts/upload_data.py
from firebase.firebase_config import db, firebase_auth
from scripts.generate_recipes import add_recipes
from scripts.generate_users import add_user


def upload_data_to_database():
    try:
        users = add_user(5)

        if not users:
            print("No users generated")
            return

        recipes_ref = db.collection("recipes")

        for user in users:
            user_auth = firebase_auth.create_user(
                email=user["email"],
                password=user["password"],
                display_name=user["username"]
            )

            db.collection("users").document(user_auth.uid).set({
                "username": user["username"],
                "email": user["email"],
                "createdAt": user["createdAt"]
            })

            recipes = add_recipes(5, user["username"])

            if not recipes:
                print("No recipes generated")
                continue

            batch = db.batch()

            for recipe in recipes:
                recipe_ref = recipes_ref.document()
                batch.set(recipe_ref, recipe)

            batch.commit()

        print("Data uploaded successfully")

    except Exception as error:
        print(str(error))